graphistry/embed_utils.py

```diff
@@ -40,18 +43,20 @@ def embed(self, proto='TransE', d=128, gamma=12):
 
         for s, r, d in zip(src, relation, dst):
             triplets.append([s, r, d])
-
         # initialize the dataloader
         dataset = EmbedDataset(entity2id, relation2id, triplets)
         train_generator = DataLoader(dataset, batch_size=32) # need autoscale batch for gpu? [AUTOML]
 
         # training loop
-        # optim (TODO: ADAM)
-        optim = torch.optim.SGD(em_mod.parameters(), lr=0.01)
+        optim = torch.optim.Adam(em_mod.parameters(), lr=0.01)
 
         # TODO: define epoch [AUTOML]
-        for _ in range(10):
 
+        epoch = trange(1000, leave=True)
```

Comment:
yes, exposing params both for users (smart overridable defaults) + later optuna is A++

Reply:
1) Eventually: add early stopping after loss fans out, etc

Reply:
1) Eventually: add early stopping after loss fans out, etc